elevation_mapping/src/sensor_processors/SensorProcessorBase.cpp

```diff
@@ -158,5 +162,37 @@ void SensorProcessorBase::removePointsOutsideLimits(
   ROS_DEBUG("removePointsOutsideLimits() reduced point cloud to %i points.", (int) pointClouds[0]->size());
 }
 
+bool SensorProcessorBase::cleanPointCloud(const pcl::PointCloud<pcl::PointXYZRGB>::Ptr pointCloud)
+{
+	pcl::PassThrough<pcl::PointXYZRGB> passThroughFilter;
+	pcl::PointCloud<pcl::PointXYZRGB> tempPointCloud;
+
+  // remove nan points
+  std::vector<int> indices;
+  pcl::removeNaNFromPointCloud(*pointCloud, tempPointCloud, indices);
+  tempPointCloud.is_dense = true;
+  pointCloud->swap(tempPointCloud);
+
+  // cutoff points with z values
+	passThroughFilter.setInputCloud(pointCloud);
+	passThroughFilter.setFilterFieldName("z");
+	passThroughFilter.setFilterLimits(sensorParameters_.at("cutoff_min_depth"),
+                                    sensorParameters_.at("cutoff_max_depth"));
+	passThroughFilter.filter(tempPointCloud);
+	pointCloud->swap(tempPointCloud);
+
+  // reduce points using VoxelGrid filter
+  double filter_size = sensorParameters_["voxelgrid_filter_size"];
```

Comment:
To stay consistent with the rest of the file, use sensorParameters_.at("voxelgrid_filter_size")